cli/src/orderbook.rs

```diff
@@ -0,0 +1,30 @@
+use asset_identifier;
+use clap::ArgMatches;
+use error::Result;
+use fmt::{Formatter, Simple};
+use stellar_client::{endpoint::{orderbook, Limit}, sync::Client};
+
+pub fn details(client: &Client, matches: &ArgMatches) -> Result<()> {
+    let endpoint = {
+        let base_asset_str = matches
+            .value_of("base_asset")
+            .expect("Base asset is a required field");
+        let counter_asset_str = matches
+            .value_of("counter_asset")
+            .expect("Counter asset is a required field");
+        let base_asset = asset_identifier::parse_identifier_from_str(base_asset_str)?;
+        let counter_asset = asset_identifier::parse_identifier_from_str(counter_asset_str)?;
+        let mut endpoint = orderbook::Details::for_asset_pair(base_asset, counter_asset);
+        if let Some(limit) = matches.value_of("limit") {
+            let limit = limit
+                .parse::<u32>()
+                .map_err(|_| String::from("Limitshould be a valid u32 integer"))?;
```

Comment:
y'know, now that you have a constructor for the Invalid error type `from_str` you can probably use it for the return values.